src/frontend/src/common/components/library/data_subview/index.tsx

```diff
@@ -207,14 +176,28 @@ const DataSubview: FunctionComponent<Props> = ({
     }
   }, [downloadFailed]);
 
-  function handleHeaderCheckboxClick() {
-    if (isHeaderIndeterminant) {
-      // clear all samples when selecting checkbox when indeterminate
-      setCheckedSamples([]);
-      setFailedSamples([]);
+  function handleHeaderCheckboxClick(
+    newPublicIds: string[],
+    newFailedIds: string[]
+  ) {
+    if (isHeaderIndeterminant || isHeaderChecked) {
+      // remove samples in current data selection when selecting checkbox when indeterminate
+      const newCheckedSamples = checkedSamples.filter(
+        (el) => !newPublicIds.includes(el)
```

Comment:
is this comparing html elements?

Reply:
no this is comparing publicId values in each TableItems (data) from the current filtered view, and removing or adding them to the checkedSamples.